File: minikasir/minikasir.java
import java.util.Scanner;

public class minikasir {
    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        System.out.print("Masukkan nama barang: ");
        String nama = input.nextLine();

        System.out.print("Masukkan harga barang: ");
        double harga = input.nextDouble();

        System.out.print("Masukkan jumlah beli: ");
        int jumlah = input.nextInt();

        double total = harga * jumlah;
        double diskon;

        // Logika sama seperti Python:
        // Jika total > 24900 → diskon = total * 0.9
        // Kalau tidak → diskon = 8
        if (total > 24900) {
            diskon = total * 0.9;
        } else {
            diskon = 8;
        }

        double totalBayar = total - diskon;

        System.out.println("\n===== STRUK PEMBELIAN =====");
        System.out.println("Nama Barang   : " + nama);
        System.out.println("Harga Satuan  : " + harga);
        System.out.println("Jumlah Beli   : " + jumlah);
        System.out.println("Total Harga   : " + total);
        System.out.println("Diskon        : " + diskon);
        System.out.println("Total Bayar   : " + totalBayar);
        System.out.println("=============================");
    }
}

File: minikasir/minikasir.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string nama;
    double harga;
    int jumlah;

    cout << "Masukkan nama barang: ";
    getline(cin, nama);

    cout << "Masukkan harga barang: ";
    cin >> harga;

    cout << "Masukkan jumlah beli: ";
    cin >> jumlah;

    double total = harga * jumlah;
    double diskon;

    // Logika sama seperti Python
    if (total > 24900) {
        diskon = total * 0.9;
    } else {
        diskon = 8;
    }

    double total_bayar = total - diskon;

    cout << "\n===== STRUK PEMBELIAN =====" << endl;
    cout << "Nama Barang   : " << nama << endl;
    cout << "Harga Satuan  : " << harga << endl;
    cout << "Jumlah Beli   : " << jumlah << endl;
    cout << "Total Harga   : " << total << endl;
    cout << "Diskon        : " << diskon << endl;
    cout << "Total Bayar   : " << total_bayar << endl;
    cout << "=============================" << endl;

    return 0;
}